sdk/index.ts

```diff
@@ -1,123 +1,64 @@
-    this.baseServer = this.validateBaseUrl(baseUrl);
-    this.authClient = new AuthApi(createConfiguration({
-      baseServer: this.baseServer,
-    }));
-    this.contractsClient = null;
-    this.customersClient = null;
-    this.eventsClient = null;
-    this.invoicesClient = null;
-    this.metersClient = null;
-    this.plansClient = null;
+    const baseServer = ConfigurationService.generateServerConfiguration(baseUrl);
+    this.configurationManager = ConfigurationService.initialize(apiKey, baseServer);
   }
 
   async login() {
-    if (this.accessToken) {
-      return;
-    }
-
-    const login = await this.authClient.login({
-      refreshToken: this.apiKey,
-    });
-
-    this.accessToken = login.accessToken;
-
-    this.generateClients();
+    await this.configurationManager.generateToken();
   }
 
   get events() {
-    this.validateIsLoggedIn(this.accessToken);
+    this.validateIsLoggedIn();
 
-    return this.eventsClient!;
+    return new EventsClient();
   }
 
   get customers() {
-    this.validateIsLoggedIn(this.accessToken);
+    this.validateIsLoggedIn();
 
-    return this.customersClient!;
+    return new CustomersClient();
   }
 
   get meters() {
-    this.validateIsLoggedIn(this.accessToken);
+    this.validateIsLoggedIn();
 
-    return this.metersClient!;
+    return new MetersClient();
   }
 
   get plans() {
-    this.validateIsLoggedIn(this.accessToken);
+    this.validateIsLoggedIn();
 
-    return this.plansClient!;
+    return new PlansClient();
   }
 
   get contracts() {
-    this.validateIsLoggedIn(this.accessToken);
+    this.validateIsLoggedIn();
 
-    return this.contractsClient!;
+    return new ContractsClient();
   }
 
   get invoices() {
-    this.validateIsLoggedIn(this.accessToken);
-
-    return this.invoicesClient!;
-  }
-
-  private validateIsLoggedIn(token: string | undefined): asserts token is string {
-    if (!token) {
-      throw new Error('Vayu client is not logged in. please call `vayu.login()` before calling this method');
-    }
-  }
+    this.validateIsLoggedIn();
 
-  private generateClients() {
-    this.validateIsLoggedIn(this.accessToken);
-    this.contractsClient = new ContractsClient(this.accessToken, this.baseServer);
-    this.customersClient = new CustomersClient(this.accessToken, this.baseServer);
-    this.eventsClient = new EventsClient(this.accessToken, this.baseServer);
-    this.invoicesClient = new InvoicesClient(this.accessToken, this.baseServer);
-    this.metersClient = new MetersClient(this.accessToken, this.baseServer);
-    this.plansClient = new PlansClient(this.accessToken, this.baseServer);
+    return new InvoicesClient();
   }
 
-  private validateBaseUrl(baseUrl?: string): BaseServerConfiguration {
-    if (!baseUrl) {
-      return server1;
-    }
-
-    const baseServer = BASE_URLS_MAP.get(baseUrl);
-
-    if (!baseServer) {
-      throw new Error(`Invalid baseUrl: ${baseUrl}`);
-    }
-
-    return baseServer;
+  private validateIsLoggedIn() {
```

Comment:
why do you need this private function if you only call to one function?

Reply:
Moved the validate is logged in to inside the client construction phase.